src/components/PopupWithForm.js

```diff
@@ -22,6 +25,27 @@ export default class PopupWithForm extends Popup {
     this.close();
   }
 
+  setLoadingState(isLoading, defaultText = "Save") {
+    console.log("we are in", this._popupElement.id);
+    if (isLoading) {
+      this._submitButtonText = "...Saving";
+      console.log(
+        "loading state is ",
+        isLoading,
+        "submitButtonText is",
+        this._submitButtonText
+      );
+    } else {
+      this._submitButtonText = defaultText;
+      console.log(
+        "loading state is ",
+        isLoading,
+        "submitButtonText is",
+        this._submitButtonText
+      );
+    }
+  }
+
```

Comment:
Needs correction: you should not close the modal in the end of the submit handler, close it in index.js after successful api response in `then` block, so the user won't need to reopen the form and fill the values again in the case of an error

Reply:
corrected